specs/WebMessageObjects.md

```diff
@@ -157,11 +234,165 @@ interface ICoreWebView2WebMessageReceivedEventArgs2 : ICoreWebView2WebMessageRec
   [propget] HRESULT AdditionalObjects(
       [out, retval] ICoreWebView2ObjectCollectionView** value);
 }
+
+[v1_enum]
+typedef enum COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND {
+  /// FileSystemHandle is for a file (i.e. FileSystemFileHandle)
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_FILE,
+  /// FileSystemHandle is for a directory (i.e. FileSystemDirectoryHandle)
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_DIRECTORY
+} COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND;
+
+[v1_enum]
+typedef enum COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION {
+  /// Read-only permission for FileSystemHandle
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READONLY,
+  /// Read and write permissions for FileSystemHandle
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READWRITE
+} COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION;
+
+[uuid(0ecf4d7d-bbf6-4320-930e-82ff6cf2d8dc), object, pointer_default(unique)]
+interface ICoreWebView2FileSystemHandle : IUnknown {
+  /// The Kind of the FileSystemHandle. It can either be a file or a directory.
+  [propget] HRESULT Kind([out, retval] COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND* kind);
+
+  /// The path to the FileSystemHandle.
+  [propget] HRESULT Path([out, retval] LPWSTR* path);
+
+  /// The permissions granted to the FileSystemHandle.
+  [propget] HRESULT Permission([out, retval] COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION* kind);
+}
+
+// Generic container of COM objects to pass to WebView2
+[uuid(d2fba648-109f-401c-b701-44fd3ddfb440), object, pointer_default(unique)]
+interface ICoreWebView2ObjectCollection : ICoreWebView2ObjectCollectionView {
+  /// Removes the object at the specified index.
+  HRESULT RemoveValueAtIndex([in] UINT32 index);
+
+  /// Inserts the object at the specified index.
+  HRESULT InsertValueAtIndex(
+      [in] UINT32 index,
+      [in] IUnknown* value);
+}
+
+[uuid(afae6f48-60d6-4b95-8b81-6f60d9c70f0b), object, pointer_default(unique)]
+interface ICoreWebView2Environment14 : ICoreWebView2Environment13 {
+  /// Create a ICoreWebView2FileSystemHandle from a path. The `path` is the path pointed by the
+  /// file and must be a syntactically correct full path but it is not checked here whether it currently
+  /// points to a file or a directory. If an invalid path is passed, an E_INVALIDARG will be returned
+  /// and the object will fail to create.

```

Comment:
I assume by "full path" we mean a fully qualified path (not a relative path). Does passing a relative path also return `E_INVALIDARG`?